packages/dx-react-scheduler-material-ui/src/templates/layouts/main-layout.jsx

```diff
@@ -0,0 +1,237 @@
+import * as PropTypes from 'prop-types';
+import Grid from '@material-ui/core/Grid';
+import { makeStyles } from '@material-ui/core/styles';
+import classNames from 'clsx';
+import { scrollingStrategy, getBorder, getBrightBorder } from '../utils';
+import { GROUPING_PANEL_VERTICAL_CELL_WIDTH } from '../constants';
+
+const useStyles = makeStyles(theme => ({
+  container: {
+    overflowY: 'auto',
+    position: 'relative',
+  },
+  stickyElement: {
+    tableLayout: 'fixed',
+    position: 'sticky',
+    overflow: 'visible',
+    background: theme.palette.background.paper,
+  },
+  header: {
+    top: 0,
+    zIndex: 2,
+  },
+  leftPanel: {
+    left: 0,
+    zIndex: 1,
+    boxSizing: 'border-box',
+    float: 'left',
+    width: ({ groupingPanelSize }) => `${theme.spacing(
+      10 + groupingPanelSize * GROUPING_PANEL_VERTICAL_CELL_WIDTH,
+    ) + 1}px`,
+  },
+  timeTable: {
+    position: 'relative',
+  },
+  mainTable: {
+    width: ({ groupingPanelSize }) => `calc(100% -
+      ${theme.spacing(10 + groupingPanelSize * GROUPING_PANEL_VERTICAL_CELL_WIDTH) + 1}px)`,
+    float: 'right',
+  },
+  fullScreenContainer: {
+    minWidth: '100%',
+    display: 'table',
+    position: 'relative',
+  },
+  autoWidth: {
+    display: 'table',
+  },
+  background: {
+    background: theme.palette.background.paper,
+  },
+  ordinaryLeftPanelBorder: {
+    borderRight: getBorder(theme),
+  },
+  brightLeftPanelBorder: {
+    borderRight: getBrightBorder(theme),
+  },
+  ordinaryHeaderBorder: {
+    borderBottom: getBorder(theme),
+  },
+  brightHeaderBorder: {
+    borderBottom: getBrightBorder(theme),
+  },
+  timeScale: {
+    width: theme.spacing(10),
+  },
+  dayScaleEmptyCell: {
+    display: 'flex',
+    alignItems: 'flex-end',
+  },
+  fullWidthTable: {
+    width: '100%',
+  },
+  leftPanelWithoutTimeScale: {
+    width: ({ groupingPanelSize }) => theme.spacing(
+      groupingPanelSize * GROUPING_PANEL_VERTICAL_CELL_WIDTH,
+    ),
+  },
+  mainTableWithoutTimeScale: {
+    width: ({ groupingPanelSize }) => `calc(100% -
+      ${theme.spacing(groupingPanelSize * GROUPING_PANEL_VERTICAL_CELL_WIDTH)}px)`,
+  },
+}));
+
+export const MainLayout = React.memo(({
+  timeScaleComponent: TimeScale,
+  dayScaleComponent: DayScale,
+  timeTableComponent: TimeTable,
+  dayScaleEmptyCellComponent: DayScaleEmptyCell,
+  groupingPanelComponent: GroupingPanel,
+  groupingPanelSize,
+  highlightDayScale,
+  setScrollingStrategy,
+  className,
+  ...restProps
+}) => {
+  const layoutRef = React.useRef(null);
+  const layoutHeaderRef = React.useRef(null);
+  const leftPanelRef = React.useRef(null);
+
+  const [isLeftBorderSet, setIsLeftBorderSet] = React.useState(false);
+  const [isTopBorderSet, setIsTopBorderSet] = React.useState(false);
+
+  React.useEffect(() => {
+    setScrollingStrategy(scrollingStrategy(
+      layoutRef.current, layoutHeaderRef.current, leftPanelRef.current,
+    ));
+  }, [layoutRef, layoutHeaderRef, leftPanelRef]);
```

Comment:
Should we  add to dependencies setScrollingStrategy and scrollingStrategy?

Reply:
It shouldn't depend on `scrollingStrategy` since this is our utils' function. But yes, I'll add setScrollingStrategy dependency